src/app/SubscriptionManager.h

```diff
@@ -0,0 +1,67 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines an interface that exposes all the public subscription management APIs.
+ *      The interface is implemented by the InteractionModelEngine to avoid creating unnecessary dependencies
+ *      Since the IMEgine has more dependency than its consummers need.
+ *      By leveraging the SubscriptionManager APIs, a consummer avoids depending on the global data model functions.
+ */
+
+#pragma once
+
+#include <lib/core/DataModelTypes.h>
+#include <lib/core/NodeId.h>
+
+namespace chip {
+namespace app {
+
+class SubscriptionManager
```

Comment:
This does not manage subscriptions (i.e. no mutations are done via this interface). It observes state. Consider rename to `SubscriptionsTracker` or `SubscriptionsInfoProvider`.